test/masq.test.js

```diff
@@ -100,29 +149,77 @@ test('should be kicked if challenge does not match', async (done) => {
   masq.requestMasqAccess()
 })
 
-test('put should reject when there is no profile selected', async () => {
-  expect.assertions(1)
-  try {
-    await masq.put('key', 'value')
-  } catch (e) {
-    expect(e.message).toBe('No profile selected')
-  }
+test('should replicate masq-profiles', async (done) => {
+  expect.assertions(4)
+  masq = new Masq()
+  await masq.init()
+
+  const challenge = masq.challenge
+  const channel = masq.channel
+  // this data is normally send by another channel
+  masqCore.receiveLink({
+    type: 'syncProfiles',
+    channel: channel,
+    challenge: challenge
+  })
+
+  masq.on('change', async msg => {
+    expect(msg.db).toBe('masq-profiles')
+    const profiles = await masq.getProfiles()
+    expect(profiles).toHaveLength(1)
+    expect(profiles[0].username).toBe('bob')
+    expect(profiles[0].id).toBeDefined()
+    const sw = masqCore.getSyncProfilesSw()
+    const hub = masqCore.getSyncProfilesHub()
+    sw.close()
+    hub.close()
+    done()
+  })
+
+  masq.requestMasqAccess()
 })
 
-test('get should reject when there is no profile selected', async () => {
+test('should fail when there is no profile selected', async () => {
   expect.assertions(1)
   try {
-    await masq.get('key')
+    masq.exchangeDataHyperdbKeys('app')
   } catch (e) {
     expect(e.message).toBe('No profile selected')
   }
 })
 
-test('del should reject when there is no profile selected', async () => {
+test('should exchange key and authorize local key if challenge matches', async (done) => {
   expect.assertions(1)
-  try {
-    await masq.del('key')
-  } catch (e) {
-    expect(e.message).toBe('No profile selected')
-  }
+  const profiles = await masq.getProfiles()
+  console.log(profiles)
+  // We check which profile corresponds to the current user
+  masq.setProfile(profiles[0].id)
+
+  const challenge = masq.challenge
+  const channel = masq.channel
+  const appName = 'app1'
+  // this data is normally send by another channel
+  masqCore.receiveLink({
+    type: 'syncData',
+    channel: channel,
+    challenge: challenge,
+    appName: appName
+  })
+
+  masqCore.on('changeDataHello', async msg => {
+    await wait()
+    // check if item is replicated in masq
+    const res = await masqCore.getItem(appName, '/hello')
+    expect(res).toEqual({ data: 'world' })
+    const sw = masqCore.getDataSw(appName)
+    const hub = masqCore.getDataHub(appName)
+    sw.close()
```

Comment:
The test shouldn't be in charge of handling swarms and hubs, the Masq mock should